### PR TITLE
Map fixes and new method for radius search

### PR DESCRIPTION
**Pull request by joeybrk372**
New Radius Search Method
- Removed context menu to load results within a radius on right/ctrl+click of the map.
- Geolocation properties now have a "Search by Radius..." extra that will open the map and start radius selection

Bug Fixes
- markers no longer snap back to original geolocation after editing/publishing new geolocation properties
- markers now appear when adding a geolocation property to an already existing workspace vertex
- markers now leave and enter clusters properly on updates
- map now handles case where a geolocation property is deleted

- [x] @sfeng88 @rygim @jharwig 
- [x] @srfarley @joeferner @mwizeman 
- [x] @EvanOxfeld @joeybrk372 @dsingley

### CODE REVIEW

**Comment by srfarley** [comment]
@joeybrk372 I tried out this brach. These are nice improvements. Should we change "Search by Radius" to "Search on Map" or "Search by Radius on Map" to make it more clear that the map will open?

File: web/war/src/main/webapp/js/map/map.js [hunk 1]
```diff
@@ -307,9 +308,34 @@ define([
...
+                    })
+                } else if (feature.id.indexOf(vertex.id) !== -1) {
+                    previousFeatures.push(feature);
+                }
+
+            });
 
-            if (!geoLocations || geoLocations.length === 0) return;
+            if (!geoLocations || geoLocations.length === 0) {
+                if (previousFeatures) {
```
Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/map/map.js]: This should be `previousFeatures.length` -- previousFeatures is initialized to `[]` on line 312, and an empty array is truthy.

File: web/war/src/main/webapp/js/map/map.js [hunk 2]
```diff
@@ -380,6 +412,8 @@ define([
 
                                 if (markers.length) {
                                     markers = self.findOrCreateMarkers(map, vertex);
+                                } else if (!markers.length && inWorkspace) {
+                                    markers = self.findOrCreateMarkers(map, vertex);
```
Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/map/map.js]: `markers = self.findOrCreateMarkers(map, vertex);` appears 3 different times. Can the assignment of `markers` be simplified?

**Comment by joeybrk372** [comment]
Good point @srfarley, changed to Search on Map

**Comment by srfarley** [comment]
:+1:

**Comment by srfarley** [comment]
@jharwig Are you OK with merging?

**Pull request merged**